code/API_definitions/qod-api.yaml

```diff
@@ -222,6 +222,363 @@ paths:
+              examples:
+                Generic400:
+                  summary: Schema validation failed
+                  value: 
+                    code: INVALID_INPUT
+                    message: "Schema validation failed at  ..."
+                NameRequired:
+                  summary: name is required
+                  value: 
+                    code: INVALID_INPUT
+                    message: "Expected property is missing: name"
+                StatusRequired:
+                  summary: status is required
+                  value: 
+                    code: INVALID_INPUT
+                    message: "Expected property is missing: status"
+                minBandwidthRequired:
+                  summary: minBandwidth
+                  value: 
+                    code: INVALID_INPUT
+                    message: "Expected property is minBandwidth"
+                maxBandwidthRequired:
+                  summary: maxBandwidth
+                  value: 
+                    code: INVALID_INPUT
+                    message: "Expected property is maxBandwidth"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "409":
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorInfo"
+              example:
+                code: CONFLICT
+                message: "Another QoS Profile is created with the same name"
+        "500":
+          description: Server error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorInfo"
+              example:
+                code: INTERNAL
+                message: "QoS Profile could not be created"
+        "503":
+          $ref: "#/components/responses/Generic503"
+    get:
+      tags:
+        - QoS Profiles
+      summary: "Get All QoS Profiles"
+      operationId: getQosProfiles
+      responses:
+        "200":
+          description: Contains information about QoS Profiles
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  allOf:
+                    - $ref: "#/components/schemas/QosProfile"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/QosProfilesNotFound404"
+        "500":
+          description: Server error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorInfo"
+              example:
+                code: INTERNAL
+                message: "Could not get QoS Profiles"
+        "503":
+          $ref: "#/components/responses/Generic503"
+  /qos-profiles/{profile-id}:
```

Comment:
I think it would be best to change `profile-id` or `qos-profile-id`. Inside of the resource itself the attribute is `id`, but outside of the resource it's best to call it `the-name-of-the-resource-id`. This just ensures that you don't end up with naming conflicts and confusion. I'm sure there will be other `profiles` in this API so `profile-id` might become ambiguous.